### ISSUE TITLE
euclidean_lossx_layer有重大BUG

### ISSUE THREAD

**Issue by YaqiLV** [daikankan/mtcnn#2]
这个loss层会导致roi和pts的loss大的离谱，网络不收敛，希望你在看看哪里写错了，我参考了这个https://github.com/BVLC/caffe/pull/5250/commits/2ba99ef81f87355930891b1b3f14f183a3055806，现在结果是正常的

**Comment by daikankan** [comment]
调整学习率试试，我自己倒是还有一个问题困惑我，参见https://github.com/dlunion/CCDL/issues/7

**Comment by YaqiLV** [comment]
我修改了两个点：
1.每个mini-batchsize，num()应该是有效样本的个数，而不是batchsize。
Dtype loss = dot / bottom[0]->num()/ Dtype(2);

2.判断是ignore_label的位置，除了loss误差置0，还把输出置为ignore_label，让这一样本不会反向传播。
diff__data[i] = 0;
output_data[i] = ignore_label_;

不过修改以后，分类的loss比回归大了10倍，感觉还是不对啊

**Comment by YaqiLV** [comment]
@daikankan  我的做法还是不对，你找到问题了吗，应该怎么改

**Comment by lbwang2006** [comment]
@YaqiLYU 训练pts有啥经验？直接train得不到很好的效果啊

**Comment by daikankan** [comment]
＠ａｌｌ，主要是最近一直没空，我之前还给作者发了邮件，探讨一些细节，感觉里面可能还有很多技巧，我一时半会儿验证不了，各位，多交流

**Comment by daikankan** [comment]
@YaqiLYU @lbwang2006 主要是最近一直没空，我之前还给作者发了邮件，探讨一些细节，感觉里面可能还有很多技巧，我一时半会儿验证不了，各位，多交流，包括代码逻辑，训练样本，超参

**Comment by lbwang2006** [comment]
@daikankan 损失函数和学习率参数设置都试了下，效果都不好啊，有啥特别的技巧吗

**Comment by githublet** [comment]
你好，我尝试了你的方法，效果不怎么好呀，参数什么的有什么好的建议嘛？

**Comment by WuFengKeDuo** [comment]
@YaqiLYU 请问你训练的Mtcnn能达到作者的水平吗现在
